src/components/shared/RankingShared.tsx

```
@@ -0,0 +1,152 @@
+  variant: RankingVariant,
+  position: number
+): string => {
+  const positionKey = Math.max(1, Math.min(position, 3)) as 1 | 2 | 3;
+  return CARD_BACKGROUND_CLASSES[variant][positionKey];
+};
+
+/**
+ * Generic ranking card wrapper — renders a header (title + icon badge)
+ * followed by a list of items using the provided render function.
+ */
+export interface BaseRankingCardProps<T>
```

ESLint

[error] Replace `T>` with `⏎··T,`

(prettier/prettier)